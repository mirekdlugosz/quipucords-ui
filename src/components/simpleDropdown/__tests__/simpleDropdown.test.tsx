import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { shallowComponent } from '../../../../config/jest.setupTests';
import { SimpleDropdown } from '../simpleDropdown';

describe('SimpleDropdown', () => {
  it('should render a basic component', async () => {
    const props = {
      label: 'lorem ipsum'
    };
    const component = await shallowComponent(<SimpleDropdown {...props} />);
    expect(component).toMatchSnapshot('basic');
  });

  it('should have a dropdown label', async () => {
    const props = {
      label: 'lorem ipsum'
    };
    const component = await shallowComponent(<SimpleDropdown {...props} />);
    expect(component.querySelector('button > span')).toMatchSnapshot('label');
  });

  it('should have a ouia id', async () => {
    const props = {
      label: 'lorem ipsum',
      menuToggleOuiaId: 'lorem_ipsum'
    };
    const component = await shallowComponent(<SimpleDropdown {...props} />);
    expect(component.querySelector('button')).toMatchSnapshot('button_ouiaid');
  });

  it('should have dropdown items with ouia id', async () => {
    const user = userEvent.setup();

    const props = {
      label: 'lorem ipsum',
      menuToggleOuiaId: 'lorem_ipsum',
      dropdownItems: [
        { item: 'dolor', ouiaId: 'ouia-dolor' },
        { item: 'sit', ouiaId: 'ouia-sit' }
      ]
    };

    const { asFragment } = render(<SimpleDropdown {...props} />);

    await user.click(screen.getByText(props.label));
    expect(asFragment()).toMatchSnapshot('expanded_ouiaid');
  });

  it('should display items and allow selecting after toggle is clicked', async () => {
    const user = userEvent.setup();
    const onMockSelect = jest.fn();

    const props = {
      label: 'Lorem ipsum',
      onSelect: onMockSelect,
      dropdownItems: [{ item: 'dolor' }, { item: 'sit' }]
    };

    const { asFragment } = render(<SimpleDropdown {...props} />);

    await user.click(screen.getByText(props.label));
    expect(asFragment()).toMatchSnapshot('expanded');

    await user.click(screen.getByText('dolor'));
    expect(onMockSelect.mock.calls).toMatchSnapshot('select');
  });
});
