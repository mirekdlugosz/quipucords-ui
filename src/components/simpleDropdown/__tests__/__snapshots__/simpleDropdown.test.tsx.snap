// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleDropdown should display items and allow selecting after toggle is clicked: expanded 1`] = `
<DocumentFragment>
  <button
    aria-expanded="true"
    aria-label="Dropdown menu"
    class="pf-v5-c-menu-toggle pf-m-expanded"
    type="button"
  >
    <span
      class="pf-v5-c-menu-toggle__text"
    >
      Lorem ipsum
    </span>
    <span
      class="pf-v5-c-menu-toggle__controls"
    >
      <span
        class="pf-v5-c-menu-toggle__toggle-icon"
      >
        <svg
          aria-hidden="true"
          class="pf-v5-svg"
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 320 512"
          width="1em"
        >
          <path
            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
          />
        </svg>
      </span>
    </span>
  </button>
  <div
    class="pf-v5-c-menu"
    data-ouia-component-id="OUIA-Generated-Dropdown-4"
    data-ouia-component-type="PF5/Dropdown"
    data-ouia-safe="true"
    data-popper-escaped="true"
    data-popper-placement="bottom-start"
    data-popper-reference-hidden="true"
    style="position: absolute; left: 0px; top: 0px; z-index: 9999; opacity: 1; transition: opacity 0ms cubic-bezier(.54, 1.5, .38, 1.11); min-width: 0px; transform: translate(0px, 0px);"
  >
    <div
      class="pf-v5-c-menu__content"
    >
      <ul
        class="pf-v5-c-menu__list"
        role="menu"
      >
        <li
          class="pf-v5-c-menu__list-item"
          data-ouia-component-id="OUIA-Generated-DropdownItem-3"
          data-ouia-component-type="PF5/DropdownItem"
          data-ouia-safe="true"
          role="none"
        >
          <button
            class="pf-v5-c-menu__item"
            role="menuitem"
            tabindex="0"
            type="button"
          >
            <span
              class="pf-v5-c-menu__item-main"
            >
              <span
                class="pf-v5-c-menu__item-text"
              >
                dolor
              </span>
            </span>
          </button>
        </li>
        <li
          class="pf-v5-c-menu__list-item"
          data-ouia-component-id="OUIA-Generated-DropdownItem-4"
          data-ouia-component-type="PF5/DropdownItem"
          data-ouia-safe="true"
          role="none"
        >
          <button
            class="pf-v5-c-menu__item"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <span
              class="pf-v5-c-menu__item-main"
            >
              <span
                class="pf-v5-c-menu__item-text"
              >
                sit
              </span>
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SimpleDropdown should display items and allow selecting after toggle is clicked: select 1`] = `
[
  [
    "dolor",
  ],
]
`;

exports[`SimpleDropdown should have a dropdown label: label 1`] = `
<span
  class="pf-v5-c-menu-toggle__text"
>
  lorem ipsum
</span>
`;

exports[`SimpleDropdown should have a ouia id: button_ouiaid 1`] = `
<button
  aria-expanded="false"
  aria-label="Dropdown menu"
  class="pf-v5-c-menu-toggle pf-m-disabled"
  data-ouia-component-id="lorem_ipsum"
  disabled=""
  type="button"
>
  <span
    class="pf-v5-c-menu-toggle__text"
  >
    lorem ipsum
  </span>
  <span
    class="pf-v5-c-menu-toggle__controls"
  >
    <span
      class="pf-v5-c-menu-toggle__toggle-icon"
    >
      <svg
        aria-hidden="true"
        class="pf-v5-svg"
        fill="currentColor"
        height="1em"
        role="img"
        viewBox="0 0 320 512"
        width="1em"
      >
        <path
          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
        />
      </svg>
    </span>
  </span>
</button>
`;

exports[`SimpleDropdown should have dropdown items with ouia id: expanded_ouiaid 1`] = `
<DocumentFragment>
  <button
    aria-expanded="true"
    aria-label="Dropdown menu"
    class="pf-v5-c-menu-toggle pf-m-expanded"
    data-ouia-component-id="lorem_ipsum"
    type="button"
  >
    <span
      class="pf-v5-c-menu-toggle__text"
    >
      lorem ipsum
    </span>
    <span
      class="pf-v5-c-menu-toggle__controls"
    >
      <span
        class="pf-v5-c-menu-toggle__toggle-icon"
      >
        <svg
          aria-hidden="true"
          class="pf-v5-svg"
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 320 512"
          width="1em"
        >
          <path
            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
          />
        </svg>
      </span>
    </span>
  </button>
  <div
    class="pf-v5-c-menu"
    data-ouia-component-id="OUIA-Generated-Dropdown-3"
    data-ouia-component-type="PF5/Dropdown"
    data-ouia-safe="true"
    data-popper-escaped="true"
    data-popper-placement="bottom-start"
    data-popper-reference-hidden="true"
    style="position: absolute; left: 0px; top: 0px; z-index: 9999; opacity: 1; transition: opacity 0ms cubic-bezier(.54, 1.5, .38, 1.11); min-width: 0px; transform: translate(0px, 0px);"
  >
    <div
      class="pf-v5-c-menu__content"
    >
      <ul
        class="pf-v5-c-menu__list"
        role="menu"
      >
        <li
          class="pf-v5-c-menu__list-item"
          data-ouia-component-id="ouia-dolor"
          data-ouia-component-type="PF5/DropdownItem"
          data-ouia-safe="true"
          role="none"
        >
          <button
            class="pf-v5-c-menu__item"
            role="menuitem"
            tabindex="0"
            type="button"
          >
            <span
              class="pf-v5-c-menu__item-main"
            >
              <span
                class="pf-v5-c-menu__item-text"
              >
                dolor
              </span>
            </span>
          </button>
        </li>
        <li
          class="pf-v5-c-menu__list-item"
          data-ouia-component-id="ouia-sit"
          data-ouia-component-type="PF5/DropdownItem"
          data-ouia-safe="true"
          role="none"
        >
          <button
            class="pf-v5-c-menu__item"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <span
              class="pf-v5-c-menu__item-main"
            >
              <span
                class="pf-v5-c-menu__item-text"
              >
                sit
              </span>
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SimpleDropdown should render a basic component: basic 1`] = `
<Dropdown
  isOpen={false}
  onOpenChange={[Function]}
  onSelect={[Function]}
  toggle={[Function]}
>
  <DropdownList />
</Dropdown>
`;
